### PR TITLE
#1326: Add clickhouse as a VectorDb.

### PR DESCRIPTION
**Pull request by vodkar**
Resolves: #1326

### CODE REVIEW

**Comment by ysolanky** [comment]
Thank you for your contribution @vodkar ! This is a great PR. Requested just one small change

File: pyproject.toml [hunk 1]
```diff
@@ -102,7 +102,7 @@ module = [
   "botocore.*",
   "bs4.*",
   "chromadb.*",
-  "clip.*",
+  "clickhouse_connect.*",
```
Comment (manthanguptaa) [inline pyproject.toml:105]: We want the user to install it on their own when they need it. Don't need it here

File: scripts/install.sh [hunk 2]
```diff
@@ -21,6 +21,9 @@ main() {
 
   print_heading "Installing phidata with [all] extras"
   pip install --editable "${REPO_ROOT}[all]"
+
```
[the remark this answers is not in the document]
Comment (vodkar) [reply scripts/install.sh]: Removed this

**Comment by vodkar** [comment]
> Thank you for your contribution @vodkar ! The `ClickhouseDb` class is not able to insert the pdf documents in the cookbook example you shared. Sharing the debug logs:
> 
> ```
> DEBUG    Debug logs enabled                                                                                                       agent.py:245
> INFO     Dropping collection                                                                                                       agent.py:77
> DEBUG    Checking if table exists: recipe_documents                                                                         clickhousedb.py:73
> DEBUG    Deleting table: recipe_documents                                                                                  clickhousedb.py:301
> INFO     Creating collection                                                                                                       agent.py:80
> DEBUG    Checking if table exists: recipe_documents                                                                         clickhousedb.py:73
> DEBUG    Creating Database: ai                                                                                              clickhousedb.py:88
> DEBUG    Creating table: recipe_documents                                                                                   clickhousedb.py:95
> INFO     Loading knowledge base                                                                                                    agent.py:83
> INFO     Reading: https://phi-public.s3.amazonaws.com/recipes/ThaiRecipes.pdf                                                        pdf.py:69
> DEBUG    Inserted 0 documents                                                                                              clickhousedb.py:212
> INFO     Added 0 documents to knowledge base    
> ```

Fixed

**Comment by manthanguptaa** [comment]
The integration is working perfectly! If you can add the `filters` column to make it consistent with PgVector database that would be good. Thank you so much for this amazing PR

**Comment by vodkar** [comment]
> The integration is working perfectly! If you can add the `filters` column to make it consistent with PgVector database that would be good. Thank you so much for this amazing PR

Filters in `PgVector2` is not used anywhere. There is a filter usage in `search()` but it seems different from `PgVector`.

I can just add `filters` field in table creation, is it ok?

**Review by manthanguptaa — APPROVED** [review]
LGTM! Thanks for your patience

**Review by ysolanky — APPROVED** [review]

**Pull request merged**